ibis/util.py

```diff
@@ -534,5 +555,86 @@ def seq_eq(
         Whether the expressions of `lefts` and `rights` are equal
     """
     return len(lefts) == len(rights) and all(
-        left.equals(right, cache=cache) for left, right in zip(lefts, rights)
+        _try_eq(left, right, cache=cache) for left, right in zip(lefts, rights)
     )
+
+
+def to_dag(expr: ir.Expr) -> Graph:
```

Comment:
Can we rename it to `to_operation_dag()`?

We could reuse it in the future to incrementally work with pure operation dags, so we can eventually remove the intermediate expressions.

Reply:
Can you cover these functions with tests?

Reply:
This code gets hit in all the repr tests, I'm not sure there's much point to it. Are there specific cases you're concerned about?